addon/index.js

```diff
@@ -12,7 +12,64 @@
  * ```
  */
 export function proto(target, key, desc) {
-  desc.value = desc.initializer();
+  desc.value = desc.initializer.apply(target);
   desc.initializer = undefined; // unset the initializer
   return desc;
 }
+
+/**
+ * Decorator which merges the value of the class field with the value
+ * of the field on the superclass.
+ *
+ * ```js
+ * class Foo {
+ *   baz = { prop1: 123 };
+ * }
+ *
+ * class Bar {
+ *   @concatenated baz = { prop2: 456 };
+ * }
+ *
+ * let bar = new Bar();
+ * console.log(bar.baz); // { prop1: 123, prop2: 456 }
+ * ```
+ */
+export function merged(target, key, desc) {
+  const valuesToMerge = desc.initializer();
+
+  desc.initializer = function() {
+    const merged = {};
+    const originalValues = this[key];
+
+    return Object.assign(merged, originalValues, valuesToMerge);
+  }
+
+  return desc;
+}
+
+/**
+ * Decorator which concatenates the value of the class field with the value
+ * of the field on the superclass.
+ *
+ * ```js
+ * class Foo {
+ *   baz = [123];
+ * }
+ *
+ * class Bar {
+ *   @concatenated baz = [456];
+ * }
+ *
+ * let bar = new Bar();
+ * console.log(bar.baz); // [123, 456]
+ * ```
+ */
+export function concatenated(target, key, desc) {
+  const valuesToConcat = desc.initializer();
```

Comment:
Same as above, stash off the original initializer and then invoke it inside the new `desc.initialier`